ProcessQueue/__init__.py

```diff
@@ -266,6 +267,273 @@ def process_failure(api_url, scan_report_id, headers):
+                    headers=headers,
+                    data=json.dumps(concepts_to_post),
+                )
+        print("CONCEPTS SAVE STATUS >>>", concept_response.status_code,
+                    concept_response.reason, flush=True)
+
+        concept_content = json.loads(concept_response.content.decode("utf-8"))
+        concept_response_content += concept_content
+
+        print("POST concepts all finished", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
+
+def get_value_concepts(values,content_type,api_url,headers):
+    names_list=[]
+    name_ids = {}
+    for value in values:
+        names_list.append(value['value'])
+        name_ids[value['value']] = value['id']
+    get_value_concept_ids= requests.get(
+            url=f"{api_url}scanreportconceptsfilter/?content_type={content_type}&fields=object_id,concept",
+            headers=headers,
+    )
+    value_concept_ids=json.loads(get_value_concept_ids.content.decode("utf-8"))
+    value_id_to_concept_map ={str(element.get("object_id", None)): str(element.get("concept", None)) for element in value_concept_ids}
+
+    print("VALUE TO CONCEPT MAP DICT",value_id_to_concept_map)
+
+    ids=list(value_id_to_concept_map.keys())
+    names=", ".join(map(str,names_list))
+    paginatedIds = paginate_chars(ids,names)
+    print("VALUE names list",names)
+    print("VALUE paginated ids",paginatedIds)
+    value_names = []
+    for Ids in paginatedIds:
+        ids_to_get = ",".join(map(str, Ids))
+        get_value_names=requests.get(
+        url=f"{api_url}scanreportvaluesfilter/?id__in={ids_to_get}&value__in={names}&fields=id,value,scan_report_field",
+        headers=headers,
+        )
+        fieldsV=json.loads(get_value_names.content.decode("utf-8"))
+        value_names.append(fieldsV)
+    value_names = [item for sublist in value_names for item in sublist]
+    print("Unfiltered values",value_names)
+
+    ###
+    get_scan_reports = requests.get(
+            url=f"{api_url}scanreports/",
+            headers=headers,
+        )
+    # get active scanreports and map them to fields. Remove any fields in archived reports
+    scanreports =json.loads(get_scan_reports.content.decode("utf-8"))
+    active_reports = [item for item in scanreports if item["hidden"] == False ]
+    active_reports_map = {str(element.get("id", None)): True for element in active_reports}
+    # get fields from values and repeat the process
+    fieldIds = set([item['scan_report_field'] for item in value_names])
+    paginatedFieldIds = paginate_chars(fieldIds,"")
+
+    fields = []
+    for Ids in paginatedFieldIds:
+        ids_to_get = ",".join(map(str, Ids))
+        
+        get_value_fields = requests.get(
+            url=f"{api_url}scanreportfieldsfilter/?id__in={ids_to_get}",
+            headers=headers,
+        )
+        fieldsV=json.loads(get_value_fields.content.decode("utf-8"))
+        fields.append(fieldsV)
+        
+    fields = [item for sublist in fields for item in sublist]
```

Comment:
We do this (and the `set()` operation) multiple times - please abstract out to a `flatten()` function.